### ISSUE TITLE
Any tweak to have NA values in training Y matrix?

### ISSUE THREAD

**Issue by gtikhonov** [TheoreticalEcology/s-jSDM#161]
Hi @MaximilianPi 

Thanks again for the amazing package!

Some of the desired predictive tasks in our analysis require that we mask part of the training community data. As far as I comprehend, inclusion of NA values to the `Y` argument of the `sjSDM(...)` function is not possible. Yet, from the point of the statistical model formulation such extension is trivial. I am wondering if you could advice some tweak how we can achieve the equivalent of including the information on missingness in entries of Y.

**Comment by MaximilianPi** [comment]
Hi @gtikhonov 

No, statistically it is not a problem. However, I need to mask the NA values in the likelihood, which requires me to do some coding. There is no actual workaround for the users. I will try to submit a pull request to the development version asap.

**Comment by damianobaldan** [comment]
Hi @MaximilianPi 

Is there an update on this issue? Such a development would be interesting for my current project too.

**Comment by MaximilianPi** [comment]
Hi @damianobaldan @gtikhonov 

I have added support for binomial (logit/probit) in the development branch, but most of the downstream functions (e.g., ANOVA/VP) are not yet supported. I don't know when I will find the time to implement everything and add thorough unit tests, so please be careful. Simple model fitting and predictions should suffice. If you need a specific downstream method, I will try to add it ASAP.

Best,
Max

**Comment by damianobaldan** [comment]
Hi @MaximilianPi 

Thank you for picking up. The plan is to use sjSDM to fit biomass data, using gaussian link on log-transformed data. I would like to mask 0 values to avoid fitting zero-inflated data. Next, I would like to use the anova and the internalStructure function to get sites and species R2 for the spatial, environmental and codistribution component. 

All the best, 
Damiano

**Comment by gtikhonov** [comment]
@MaximilianPi could you check my comment on your modifications in 5cdefba3ed1206ee5aaf7ac09ab1e516592dfee7 please?

**Comment by MaximilianPi** [comment]
Hi @gtikhonov 

Thanks! Check my commit, first I fill the NA in a copy of Y to prevent numerical problems when we calculate the logLik, and then I mask_fill the likelihood to detach the gradients...it should work, although it is not the most elegant approach.
